arangod/RestHandler/RestStatusHandler.cpp

```diff
@@ -274,3 +285,26 @@ RestStatus RestStatusHandler::executeOverview() {
   generateResult(rest::ResponseCode::OK, result.slice());
   return RestStatus::DONE;
 }
+
+RestStatus RestStatusHandler::executeMemoryProfile() {
+#if defined(USE_MEMORY_PROFILE)
+  std::string const outDirectory = TRI_GetTempPath();
+  std::string const fileName = FileUtils::buildFilename(outDirectory, "profile.out");
```

Comment:
I think it should be changed further, as `fileName` will the same for all threads that enter this method, so it is still not thread-safe. I think we should produce a random filename every time or add a mutex.

Reply:
done